ext/hash/hash.c

```diff
@@ -414,8 +414,8 @@ PHP_FUNCTION(hash_init)
 
 #define PHP_HASHCONTEXT_VERIFY(func, hash) { \
 	if (!hash->context) { \
-		php_error(E_WARNING, "%s(): supplied resource is not a valid Hash Context resource", func); \
-		RETURN_NULL(); \
+		zend_throw_error(NULL, "%s(): supplied resource is not a valid Hash Context resource", func); \
```

Comment:
Should this be a TypeError?  See discussion on #4643

Reply:
Not really the same case, more a bad error message. It should probably say something along the lines of "can't reuse an already finalized HashContext" or so.

Reply:
Gotcha. 👍